build/grafana-frontend.yaml

```diff
@@ -0,0 +1,44 @@
+# Copyright 2023 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Connects to Prometheus frontend UI: https://cloud.google.com/stackdriver/docs/managed-prometheus/query#ui-prometheus
+
+service:
+  port: 3000
+tolerations:
+- key: "agones.dev/agones-metrics"
+  operator: "Equal"
+  value: "true"
+  effect: "NoExecute"
+affinity:
+  nodeAffinity:
+    preferredDuringSchedulingIgnoredDuringExecution:
+    - weight: 1
+      preference:
+        matchExpressions:
+          - key: agones.dev/agones-metrics
+            operator: Exists
+sidecar:
+    dashboards:
+        enabled: true
+plugins: grafana-piechart-panel
+datasources:
+  datasources.yaml:
+    apiVersion: 1
+    datasources:
+    - name: Prometheus
+      type: prometheus
+      url: http://frontend.metrics.svc.cluster.local:9090/
```

Comment:
It looks like this line is the only difference from `build/grafana.yaml`. I'm wondering about maintainability in the medium to long term of the two files and how we can prevent drift. Maybe something to file a ticket about, but I don't think it should block this PR from merging.

Reply:
Yeah, I spent a little while on this before putting the PR up and didn't come up with a great answer. Part of the problem is that I can't figure out a one-liner `--set foo=bar` that will override this exact field. I _think_ I could do this as a third merge, i.e. something like `-f build/grafana.yaml -f build/grafana-override-frontend.yaml`, but I think that's fussier than what I have here.

Another approach I looked at was just renaming the service that comes out of the frontend install. At the end of the day, the frontend itself is a weird thing and Grafana should just have native support for the PromQL queries to Managed Prometheus - the frontend is just acting as an auth proxy right now. I think I'm going to follow up on that one internally first.

Reply:
Filed #2994.